### PR TITLE
MSCNG: add initial decrypt support

### PR DESCRIPTION
**Pull request by vmiklos**
Just rsa-1_5 / aes256-cbc as a start.

### CODE REVIEW

File: src/mscng/certkeys.c [hunk 1]
```diff
@@ -1059,13 +1059,33 @@ xmlSecMSCngKeyDataRsaGetType(xmlSecKeyDataPtr data) {
...
+}
+
 static xmlSecSize
 xmlSecMSCngKeyDataRsaGetSize(xmlSecKeyDataPtr data) {
     xmlSecAssert2(xmlSecKeyDataCheckId(data, xmlSecMSCngKeyDataRsaId), 0);
 
-    xmlSecNotImplementedError(NULL);
-
-    return(0);
+    return(xmlSecMSCngKeyDataGetSize(data));
```
Comment (lsh123) [inline src/mscng/certkeys.c]: Just curious, why do you need a separate function? Do you expect to use it somewhere else?

File: src/mscng/kt_rsa.c [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+/**************************************************************************
+ *
+ * Internal MSCNG RSA PKCS1 CTX
+ *
+ *************************************************************************/
+typedef struct _xmlSecMSCngRsaPkcs1OaepCtx xmlSecMSCngRsaPkcs1OaepCtx, *xmlSecMSCngRsaPkcs1OaepCtxPtr;
+
+struct _xmlSecMSCngRsaPkcs1OaepCtx {
+    xmlSecKeyDataPtr data;
+    int ctxInitialized;
```
Comment (lsh123) [inline src/mscng/kt_rsa.c]: I don't think you need it and I don't see it used.

File: src/mscng/kt_rsa.c [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+    xmlSecAssert2(xmlSecMSCngRsaPkcs1OaepCheckId(transform), -1);
+    xmlSecAssert2(xmlSecTransformCheckSize(transform, xmlSecMSCngRsaPkcs1OaepCtx), -1);
+
+    ctx = xmlSecMSCngRsaPkcs1OaepGetCtx(transform);
+    xmlSecAssert2(ctx != NULL, -1);
+
+    /* initialize */
+    memset(ctx, 0, sizeof(xmlSecMSCngRsaPkcs1OaepCtx));
+
+    if(xmlSecTransformCheckId(transform, xmlSecMSCngTransformRsaPkcs1Id)) {
```
Comment (lsh123) [inline src/mscng/kt_rsa.c]: May be 

```
if (!xmlSecTransformCheckId(...)) {
    xmlSecInvalidTransformError(...);
   return(-1);
}
// happy path
```

File: src/mscng/kt_rsa.c [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+        ret = xmlSecBufferSetData(out, xmlSecBufferGetData(in), inSize);
+        if(ret < 0) {
+            xmlSecInternalError2("xmlSecBufferSetData",
+                xmlSecTransformGetName(transform), "size=%d", inSize);
+            return(-1);
+        }
+
+        dwInLen = inSize;
+        dwBufLen = outSize;
+	hPubKey = xmlSecMSCngKeyDataGetPubKey(ctx->data);
```
Comment (lsh123) [inline src/mscng/kt_rsa.c]: space

File: src/mscng/certkeys.c (hunk 1 above)
Comment (vmiklos) [reply src/mscng/certkeys.c]: I think there is some value in asserting my input but still doing the real work in an external function, so later I can call that in e.g. xmlSecMSCngKeyDataDsaGetSize(). Can refer to this shared function in xmlSecMSCngKeyDataRsaKlass, but then I loose my asserts.

File: src/mscng/kt_rsa.c (hunk 2 above)
Comment (vmiklos) [reply src/mscng/kt_rsa.c]: Indeed, I used it as a placeholder when the struct was "empty" (to get the build going) and forgot to remove it when I added the `data` member.

File: src/mscng/kt_rsa.c (hunk 3 above)
Comment (vmiklos) [reply src/mscng/kt_rsa.c]: Done.

File: src/mscng/kt_rsa.c (hunk 4 above)
Comment (vmiklos) [reply src/mscng/kt_rsa.c]: Done, there were also a couple of other places in the same file where I used tabs by accident, all fixed now.

**Pull request merged**